clang/lib/Parse/ParseDecl.cpp

```diff
@@ -2071,6 +2071,29 @@ void Parser::SkipMalformedDecl() {
   }
 }
 
+void ParseImplicitDeclareTargetAttr(Decl *TargetDecl) {
+  if (TargetDecl && TargetDecl->hasAttr<OMPDeclareTargetDeclAttr>() && isa<VarDecl>(TargetDecl)) {
+    VarDecl *VarFirstDecl = cast<VarDecl>(TargetDecl);
+    Expr *Ex = VarFirstDecl->getInit()->IgnoreCasts();
+    const DeclRefExpr *DeclRef = nullptr;
+    if (Ex && isa<UnaryOperator>(Ex)) {
+      if (auto *unary = cast<UnaryOperator>(Ex)) {
+	if (unary->getOpcode() == UnaryOperator::Opcode::UO_AddrOf) {
```

Comment:
It looks like this would only handle initializers that have the address-of operator. Were you going to also handle the following:

```
int y;
#pragma omp begin declare target
int x = y;
#pragma omp end declare target 
```